roles/prune_catalog/tasks/main.yml

```
@@ -11,16 +11,18 @@
       - pc_destination_catalog | length > 0
     fail_msg: "At least one of the parameters requirements is missing"
 
-- name: "Check if image is pullable - {{ pc_source_catalog }} "
-  ansible.builtin.shell: >
-    if [ -n "$DOCKER_CONFIG" ]; then
-      skopeo inspect --no-tags \
-      --authfile "$DOCKER_CONFIG/config.json" \
-      --tls-verify=false docker://{{ pc_source_catalog }}
-    else
-      skopeo inspect --no-tags \
-      --tls-verify=false docker://{{ pc_source_catalog }}
-    fi
+- name: "Check if image is pullable - {{ pc_source_catalog }}"
+  vars:
+    docker_config_defined: "{{ DOCKER_CONFIG is defined }}"
+  ansible.builtin.command: >
+    skopeo inspect --no-tags
+    {%- if pc_allow_insecure_registry | bool %}
+    --tls-verify=false
+    {%- endif %}
+    {%- if docker_config_defined | bool %}
+    --authfile "$DOCKER_CONFIG/config.json"
+    {%- endif %}
+    docker://{{ pc_source_catalog }}
```

I'm not sure if `DOCKER_CONFIG` could be evaluated like that since is an `env` and not an ansible var.

I think this should do it:

```suggestion
- name: "Check if image is pullable - {{ pc_source_catalog }}"
  ansible.builtin.command: >
    skopeo inspect --no-tags
    {%- if pc_allow_insecure_registry | bool %}
    --tls-verify=false
    {%- endif %}
    {%- if lookup('ansible.builtin.env', 'DOCKER_CONFIG') | length %}
    --authfile "$DOCKER_CONFIG/config.json"
    {%- endif %}
    docker://{{ pc_source_catalog }}
```